lnwallet/btcwallet/btcwallet.go

```diff
@@ -1424,7 +1427,19 @@ func (b *BtcWallet) ListTransactionDetails(startHeight, endHeight int32,
 		txDetails = append(txDetails, detail)
 	}
 
-	return txDetails, nil
+	if filterFn == nil {
+		// No filter function applied.
+		return txDetails, nil
+	}
+
+	filteredTxs := []*lnwallet.TransactionDetail{}
```

Comment:
nit: use `var filteredTxs []*lnwallet.TransactionDetail` if not using a pre-initialized value.